nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PostSlack.java

```diff
@@ -82,6 +83,14 @@
                 " The property name will not be used by the processor.",
         expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
 @WritesAttribute(attribute="slack.file.url", description = "The Slack URL of the uploaded file. It will be added if 'Upload FlowFile' has been set to 'Yes'.")
+@DeprecationNotice(
+    alternatives = {PublishSlack.class},
+    reason = "This Processor is more difficult to configure than necessary, and exposes intricate nuances of the Slack API that need not be exposed. " +
+             "Additionally, it lacks some capabilities, such as responding to messages in threads. It interacts directly with the Slack API instead of " +
+             "making use of the Slack-provided SDK, which makes the Processor much more brittle. Additionally, it does not make clear the differences " +
+             "between PutSlack and PostSlack. Each provides slightly different capabilities. " +
+             "PublishSlack combines the capabilities of both Processors in a simpler configuration that makes use of Slack's standard REST API."
+)
```

Comment:
Rather than tag this on deprecated on the main branch, the deprecation should be added on the support branch. The removal of these Processors could be handled in this pull request, or in a subsequent PR.